src/vma/proto/netlink_socket_mgr.h

```diff
@@ -130,15 +136,19 @@ netlink_socket_mgr <Type>::netlink_socket_mgr(nl_data_t data_type)
 	memset(m_msg_buf, 0, m_buff_size);
 
 	// Create Socket
+
 	BULLSEYE_EXCLUDE_BLOCK_START
-	if ((m_fd = orig_os_api.socket(PF_NETLINK, SOCK_DGRAM, NETLINK_ROUTE)) < 0) {
+	m_sock = nl_socket_alloc();
+	if (m_sock == nullptr) {
 		__log_err("NL socket Creation: ");
 		return;
 	}
 
-	if (orig_os_api.fcntl(m_fd, F_SETFD, FD_CLOEXEC) != 0) {
-		__log_warn("Fail in fctl, error = %d", errno);
-	}
+	if (nl_connect(m_sock, NETLINK_ROUTE) < 0) {
+		__log_err("NL socket Connection: ");
+        nl_socket_free(m_sock);
+    }
+
```

Comment:
formatting

Reply:
please consider
```
	if (nl_connect(m_sock, NETLINK_ROUTE) < 0) {
		__log_err("NL socket Connection: ");
                nl_socket_free(m_sock);
               m_sock = nullptr;
              return;
    }
```
to be consistent and return at the end just in case good flow.

Reply:
super right. fixed. Thanks :)